test/aws/osml/model_runner/sink/test_sink.py

```diff
@@ -0,0 +1,74 @@
+#  Copyright 2023-2024 Amazon.com, Inc. or its affiliates.
+
+import unittest
+from typing import List
+
+from geojson import Feature
+
+from aws.osml.model_runner.api import SinkMode
+from aws.osml.model_runner.sink.sink import Sink
+
+
+class MockSink(Sink):
+    """
+    A mock implementation of the Sink abstract class to test the interface.
+    """
+
+    @staticmethod
+    def name() -> str:
+        return "MockSink"
+
+    @property
+    def mode(self) -> SinkMode:
+        return SinkMode.AGGREGATE
+
+    def write(self, image_id: str, features: List[Feature]) -> bool:
+        return True
+
+
+class TestSink(unittest.TestCase):
+    def setUp(self):
+        """
+        Create an instance of the MockSink class for testing.
+        """
+        self.sink = MockSink()
+
+    def tearDown(self):
+        """
+        Clean up any resources used by the tests.
+        """
+        self.sink = None
+
+    def test_sink_name(self):
```

Comment:
Testing name, mode, and write on your mock really isn't testing anything. I can see the value in testing your __str__ representation since that is actual custom code but the rest of it is just code for coverage sake.

Reply:
Removed the other tests and left str representation